specification/cognitiveservices/data-plane/AnomalyFinder/stable/v2.0/AnomalyFinder.json

```diff
@@ -0,0 +1,331 @@
+          "type": "number",
+          "format": "float",
+          "description": "Optional argument, advanced model parameter, max anomaly ratio in a time series."
+        },
+        "sensitivity": {
+          "type": "number",
+          "format": "float",
+          "description": "Optional argument, advanced model parameter, between 0-99, the lower the value is, the larger the margin value will be which means less anomalies will be accepted."
+        }
+      }
+    },
+    "Point": {
+      "type": "object",
+      "required": [
+        "timestamp",
+        "value"
+      ],
+      "properties": {
+        "timestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Timestamp of a data point. Please make sure it aligns with the midnight, and use a UTC date time string, e.g., 2017-08-01T00:00:00Z."
+        },
+        "value": {
+          "type": "number",
+          "format": "float",
+          "description": "The measurement of that point, should be float."
+        }
+      }
+    },
+    "EntireDetectResponse": {
+      "type": "object",
+      "required": [
+        "expectedValues",
+        "isAnomaly",
+        "isNegativeAnomaly",
+        "isPositiveAnomaly",
+        "lowerMargins",
+        "period",
+        "upperMargins"
+      ],
+      "properties": {
+        "period": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Frequency extracted from the series, zero means no recurrent pattern has been found."
+        },
+        "expectedValues": {
+          "type": "array",
+          "description": "ExpectedValues contain expected value for each input point. The index of the array is consistent with the input series.",
+          "items": {
+            "type": "number",
+            "format": "float"
+          }
+        },
+        "upperMargins": {
+          "type": "array",
+          "description": "UpperMargins contain upper margin of each input point. UpperMargin is used to calculate upperBoundary, which equals to expectedValue + (100 - sensitivity)*upperMargin. Anomalies in response can be filtered by upperBoundary and lowerBoundary. By adjusting sensitivity value, less significant anomalies can be filtered in client side. The index of the array is consistent with the input series.",
+          "items": {
+            "type": "number",
+            "format": "float"
+          }
+        },
+        "lowerMargins": {
+          "type": "array",
+          "description": "LowerMargins contain lower margin of each input point. LowerMargin is used to calculate lowerBoundary, which equals to expectedValue - (100 - sensitivity)*lowerMargin. Points between the boundary can be marked as normal ones in client side. The index of the array is consistent with the input series.",
+          "items": {
+            "type": "number",
+            "format": "float"
+          }
+        },
+        "isAnomaly": {
```

Comment:
I would suggest merging isAnomaly, isNegativeAnomaly and isPositiveAnomaly into a single array of numbers. This reduces the amount of network traffic/time to serialize/deserialize.

Reply:
After discussing with team members, we would not merge the three arrays into one. The reasons are:  

1. Based on feedback of existing customers,  some users needs to distinguish anomalies as positive ones or negative ones,  others only use isAnomaly as their final results.  If we combine these together, they have to understand these concept.
2. From Maintenance aspect,  we may deprecate positive and negative concept or replace it as spike or dips, or even add change point. In separate ways, we can remove or add arrays.
3. Positive and negative are additional features we provided on top of anomalies,  if we merge them together,  an incorrect judgement on positive or negative direction may make customer confuses whether it's a true anomaly or not.